source/state_representation/src/space/cartesian/CartesianState.cpp

```diff
@@ -264,8 +264,8 @@ double CartesianState::dist(const CartesianState& state, const CartesianStateVar
   if (state_variable_type == CartesianStateVariable::ORIENTATION || state_variable_type == CartesianStateVariable::POSE
       || state_variable_type == CartesianStateVariable::ALL) {
     // https://math.stackexchange.com/questions/90081/quaternion-distance for orientation
-    double inner_product = this->get_orientation().dot(state.get_orientation());
-    result += acos(2 * inner_product * inner_product - 1);
+    double argument = 2 * this->get_orientation().dot(state.get_orientation()) - 1;
```

Comment:
i'm confused, isn't this missing the square operation?

Reply:
yeah...sorry